operator/pkg/kube/operator/operator.go

```diff
@@ -0,0 +1,240 @@
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+        apiclient "k8s.io/client-go/kubernetes"
+        "k8s.io/client-go/rest"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const (
+	operatorLabel = "istio-operator"
+	maxRetries = 5
+)
+
+var (
+        runtimeScheme = runtime.NewScheme()
+        codecs        = serializer.NewCodecFactory(runtimeScheme)
+        deserializer  = codecs.UniversalDeserializer()
+)
+
+// LoadKubeConfig is a unit test override variable for loading the k8s config.
+// DO NOT USE - TEST ONLY.
+var LoadKubeConfig = clientcmd.Load
+
+// CreateInterfaceFromClusterConfig is a unit test override variable for interface create.
+// DO NOT USE - TEST ONLY.
+var CreateInterfaceFromClusterConfig = kube.CreateInterfaceFromClusterConfig
+
+// addConfigMapCallback prototype for unit testing.
+// DO NOT USE - TEST ONLY.
+type addConfigMapCallback func(*corev1.ConfigMap) error
+
+// removeConfigMapCallback prototype for unit testing.
+// DO NOT USE - TEST ONLY.
+type removeConfigMapCallback func(string) error
+
+// addConfigMapCallback is called during unit tetsts to verify this controller is working properly.
+// DO NOT USE - TEST ONLY.
+//var addConfigMapCallback = nil
+
+// rmoveConfigMapCallback is called during unit tetsts to verify this controller is working properly.
+// DO NOT USE - TEST ONLY.
+//var removeConfigMapCallback = nil
+
+// Controller is the controller implementation for ConfigMap resources.
+type Controller struct {
+        clientExt *apiextensionsclient.Clientset
+	kuberest  *rest.Config
+	namespace string
+	queue     workqueue.RateLimitingInterface
+	informer  cache.SharedIndexInformer
+}
+
+func init() {
+	corev1.AddToScheme(runtimeScheme)
+        apiextensionsv1beta1.AddToScheme(runtimeScheme)
+}
+
+// NewController returns a new operator controller.
+func NewController(
+	restConfig *rest.Config,
+	namespace string) *Controller {
+
+	kubeclientset, _ := apiclient.NewForConfig(restConfig)
+
+	clientExt, _ := apiextensionsclient.NewForConfig(restConfig)
+
+	configMapInformer := cache.NewSharedIndexInformer(
+		&cache.ListWatch{
+			ListFunc: func(opts meta_v1.ListOptions) (runtime.Object, error) {
+				opts.LabelSelector = operatorLabel + "=true"
+				return kubeclientset.CoreV1().ConfigMaps(namespace).List(opts)
+			},
+			WatchFunc: func(opts meta_v1.ListOptions) (watch.Interface, error) {
+				opts.LabelSelector = operatorLabel + "=true"
+				opts.Watch = true
+				return kubeclientset.CoreV1().ConfigMaps(namespace).Watch(opts)
+			},
+		},
+		&corev1.ConfigMap{}, 0, cache.Indexers{},
+	)
+
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+
+	controller := &Controller{
+		clientExt:	clientExt,
+		namespace:      namespace,
+		informer:       configMapInformer,
+		queue:          queue,
+	}
+
+	configMapInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
since you do not plan to put back on the queue received object, I would suggest not to use it and call directly a function to deal with ConfigMap add/delete. Lots of unneeded complexity will go away. Also you can do type assertion right here and pass to your func v1.ConfigMap object.